src/Raven.Embedded/EmbeddedServer.cs

```diff
@@ -29,26 +31,48 @@ internal EmbeddedServer()
         }
 
         private readonly Logger _logger = LoggingSource.Instance.GetLogger<EmbeddedServer>("Embedded");
-        private Lazy<Task<(Uri ServerUrl, Process ServerProcess)>> _serverTask;
+        private Lazy<Task<(Uri ServerUrl, Process ServerProcess)>>? _serverTask;
 
         private readonly ConcurrentDictionary<string, Lazy<Task<IDocumentStore>>> _documentStores = new ConcurrentDictionary<string, Lazy<Task<IDocumentStore>>>();
-        private X509Certificate2 _certificate;
 
-        private TimeSpan _gracefulShutdownTimeout;
+        private ServerOptions? _serverOptions;
 
-        public void StartServer(ServerOptions options = null)
+        public async Task<int> GetServerProcessId(CancellationToken token = default)
         {
-            options = options ?? ServerOptions.Default;
+            var server = _serverTask;
+            if (server == null)
+                throw new InvalidOperationException($"Please run {nameof(StartServer)}() before trying to use the server");
+            return (await server.Value.WithCancellation(token).ConfigureAwait(false)).ServerProcess.Id;
+        }
 
-            _gracefulShutdownTimeout = options.GracefulShutdownTimeout;
-            var startServer = new Lazy<Task<(Uri ServerUrl, Process ServerProcess)>>(() => RunServer(options));
-            if (Interlocked.CompareExchange(ref _serverTask, startServer, null) != null)
-                throw new InvalidOperationException("The server was already started");
+        public void Restart()
```

Comment:
Maybe make this method async?

Reply:
It should be called `RestartServer` to match the `StartServer` one

Reply:
I made is sync and run in background because StartServer is also that.
But it make sense that you'll want to get any errors immediately.